server.js

```diff
@@ -64,40 +69,48 @@ app.use('/', (req, res, next) => {
 // Init the nypl data api client.
 nyplApiClient();
 
-app.use('/*', initializePatronTokenAuth, getPatronData);
-app.use('/', apiRoutes);
-
-app.get('/*', (req, res) => {
-  alt.bootstrap(JSON.stringify(res.locals.data || {}));
-
+function renderReact(req, res, renderPage = false) {
   const appRoutes = (req.url).indexOf(appConfig.baseUrl) !== -1 ? routes().client : routes().server;
-
   match({ routes: appRoutes, location: req.url }, (error, redirectLocation, renderProps) => {
     if (error) {
       res.status(500).send(error.message);
     } else if (redirectLocation) {
       res.redirect(302, redirectLocation.pathname + redirectLocation.search);
     } else if (renderProps) {
-      const application = ReactDOMServer.renderToString(<RouterContext {...renderProps} />);
-      const title = DocumentTitle.rewind();
-      const iso = new Iso();
-
-      iso.add(application, alt.flush());
-      res
-        .status(200)
-        .render('index', {
-          application: iso.render(),
-          appTitle: title,
-          favicon: appConfig.favIconPath,
-          webpackPort: WEBPACK_DEV_PORT,
-          path: req.url,
-          isProduction,
-          baseUrl: appConfig.baseUrl,
-        });
+      application = ReactDOMServer.renderToString(<RouterContext {...renderProps} />);
+      const flushed = alt.flush();
+      if (renderPage) {
+        iso.add(application, flushed);
+        res
+          .status(200)
+          .render('index', {
+            application: iso.render(),
+            appTitle: title,
+            favicon: appConfig.favIconPath,
+            webpackPort: WEBPACK_DEV_PORT,
+            path: req.url,
+            isProduction,
+            baseUrl: appConfig.baseUrl,
+          });
+      }
     } else {
       res.status(404).redirect(`${appConfig.baseUrl}/`);
     }
   });
+}
+
+app.use('/*', initializePatronTokenAuth, getPatronData);
+app.use('/', apiRoutes);
+
+app.get('/*', (req, res, next) => {
+  Store.next = next;
```

Comment:
Why do you need to store this function in the store?... This seems more like a workaround than something stable.